### PR TITLE
Fixed issue [security] #18977: Improper Authorization in add role function leads to privilege escalation

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue [security] #18977: Improper Authorization in add role function leads to privilege escalation
Dev: limited to superadmin (read)

### CODE REVIEW

File: application/controllers/UserManagementController.php [hunk 1]
```diff
@@ -1163,9 +1164,8 @@ public function actionBatchApplyRoles()
             $aResults[$sItem]['title'] = '';
             $model = $this->loadModel($sItem);
             $aResults[$sItem]['title'] = $model->users_name;
-
-            //check if user is admin otherwhise change the role
-            if (intval($sItem) == 1) {  //todo REFACTORING is admin id always 1?? is there another possibility to check for admin user?
+            if (Permission::isForcedSuperAdmin($sItem)) {
```
Comment (gabrieljenik) [inline application/controllers/UserManagementController.php:1167]: Same comment as in other PR.
Not sure this is part of the ticket. Must be clarified for better testing.

**Review by gabrieljenik — APPROVED** [review]

File: application/controllers/UserManagementController.php (hunk 1 above)
Comment (Shnoulle) [reply application/controllers/UserManagementController.php:1167]: It's clear the `//todo REFACTORING is admin id always 1??`

Permission::isForcedSuperAdmin id done to replace loginid == 1 …

Comment (Shnoulle) [reply application/controllers/UserManagementController.php:1167]: https://github.com/LimeSurvey/LimeSurvey/pull/810

Comment (gabrieljenik) [reply application/controllers/UserManagementController.php:1167]: > It's clear the //todo REFACTORING is admin id always 1??

It is clear in the PR, but will the tester know that? Is it on the ticket?

**Pull request merged**